src/app/player.js

```diff
@@ -0,0 +1,471 @@
+        return (this.OOPlayer) ? this.OOPlayer.getPlayheadTime() : null;
+    }
+
+    
+    /**
+     * getDurationSec returns the duration of the asset
+     * 
+     * @returns {number} time in seconds
+     * @memberof CastPlayer
+     */
+    getDurationSec() {
+        return (this.OOPlayer) ? this.OOPlayer.getDuration() : null;
+    }
+
+    
+    /**
+     * setClosedCaptions set the closed caption language for the asset. The value
+     * needs to be one of the given by the asset info
+     * 
+     * @param {string} lang 
+     * @memberof CastPlayer
+     */
+    setClosedCaptions(lang) {
+        if (lang === '' || !lang) {
+            lang = 'none';
+        }
+        logger.info(LOG_PREFIX, "Closed captions new value:", lang);
+        this.OOPlayer.setClosedCaptionsLanguage(lang);
+    }
+
+    
+    /**
+     * getState returns the actual state of the player. This method will convert from
+     * Ooyala player states to google cast states
+     * 
+     * @returns {cast.receiver.media.PlayerState}
+     * @memberof CastPlayer
+     */
+    getState() {
+        if (this.OOPlayer) {
+            switch (this.OOPlayer.getState()) {
+                case OO.STATE.LOADING:
+                case OO.STATE.BUFFERING:
+                    this.state = cast.receiver.media.PlayerState.BUFFERING;
+                    break;
+                case OO.STATE.PLAYING:
+                    this.state = cast.receiver.media.PlayerState.PLAYING;
+                    break;
+                case OO.STATE.PAUSED:
+                    this.state = cast.receiver.media.PlayerState.PAUSED;
+                    break;
+                case OO.STATE.READY:
+                    this.state = cast.receiver.media.PlayerState.IDLE;
+                break;
+            }
+        }
+        logger.info(LOG_PREFIX,"State:", this.state);
+        return this.state;
+    }
+
+    
+    /**
+     * play executes the play method on the Ooyala player
+     * 
+     * @memberof CastPlayer
+     */
+    play() {
+        this.OOPlayer.play();
+    }
+
+    /**
+     * pause executes the pause method on the Ooyala player
+     * 
+     * @memberof CastPlayer
+     */
+    pause() {
+        this.OOPlayer.pause();
+    }
+
+    /**
+     * reset is not implemented but it was keep it to follow the full implementation of the
+     * player interface provied by Google cast sdk 
+     * 
+     * @memberof CastPlayer
+     */
+    reset() {
+    }
+
+    
+    /**
+     * seek sets the current position of the player on the asset from the beginning 
+     * 
+     * @param {number} time 
+     * @memberof CastPlayer
+     */
+    seek(time) {
+        this.OOPlayer.seek(time);
+    }
+
+    /**
+     * setVolume sets the current volume of the Ooyala player. The value must be betweeen
+     * 0 and 1, inclusive
+     * 
+     * @param {any} volume 
+     * @memberof CastPlayer
+     */
+    setVolume(volume) {
+        this.OOPlayer.setVolume(volume.level);
```

Comment:
Should we add a null check for this.OOPlayer? Should we also use max/min functions to ensure the volume is between 0 and 1? We should also add a null check for the volume parameter.